outport/firehose/firehoseIndexer.go

```diff
@@ -122,6 +131,57 @@ func (fi *firehoseIndexer) getHeaderBytes(headerHandler data.HeaderHandler) ([]b
 	return headerBytes, headerType, err
 }
 
+func getBody(bodyHandler data.BodyHandler) (*block.Body, error) {
+	if check.IfNil(bodyHandler) {
+		return &block.Body{}, nil
+	}
+
+	body, castOk := bodyHandler.(*block.Body)
+	if !castOk {
+		return nil, errCannotCastBlockBody
+	}
+
+	return body, nil
+}
+
+func getBodyOrNilIfEmpty(body *block.Body) *block.Body {
+	if len(body.MiniBlocks) == 0 {
+		return nil
+	}
+
+	return body
+}
+
+func getAlteredAccounts(accounts map[string]*outportcore.AlteredAccount) []*alteredAccount.AlteredAccount {
+	ret := make([]*alteredAccount.AlteredAccount, 0, len(accounts))
+
+	for _, acc := range accounts {
+		ret = append(ret, &alteredAccount.AlteredAccount{
```

Comment:
It seems you always have exactly len(accounts) elements, so I would do simple assignment instead of append in this case, like you did with tokens.

`alteredAccounts[i] = *alteredAccount.AlteredAccount{...}`
and also change the initialization of alteredAccount to `make([]*alteredAccount.AlteredAccount, len(accounts))`

Reply:
Right, good suggestion. Applied it.